### DISCUSSION TITLE
How to train an initial model

### DISCUSSION THREAD

**Discussion by 34j** [voicepaw/so-vits-svc-fork#579] [General]
I am not sure if it is worth training an initial model (because this repo is poorly maintained), but here is how to train an initial model.

- Prepare a Paperspace Growth account or a 3090/4090 at home etc.
- Change the config to suit your purpose. fp16 on and frequency to 22kHz may give you a somewhat reduced quality statement and faster training. (https://github.com/voicepaw/so-vits-svc-fork/discussions/303)
- Place the legacy initial model to reuse some weights if necessary. However, it may be better to train from nothing, as the legacy initial model may have been trained with non-open dataset.
- Use a dataset that includes a variety of speakers, and is very loosely licensed, such as VCTK.
- Modify the code as needed to fix weights.
- Not sure how long it will take, but expect training to take anywhere from a few hours to a few days.
- Once the training is complete, please send a PR updating the URL of the initial model.
https://github.com/voicepaw/so-vits-svc-fork/blob/7d02117799eb947a0486101b7fba82d9707d9b61/src/so_vits_svc_fork/utils.py#L87
- If you have changed the configuration, please provide the configuration file as well.
https://github.com/voicepaw/so-vits-svc-fork/tree/main/src/so_vits_svc_fork/preprocessing/config_templates
- I am not an expert in machine learning, so everything I am saying could be wrong.
(I can't provide a model because I trained on a non-open data set for personal taste, but samples are available on the [Wiki](https://github.com/voicepaw/so-vits-svc-fork/wiki).)

**Comment by SamueleLorefice** [comment]
> Place the legacy initial model to reuse some weights if necessary. However, it may be better to train from nothing, as the legacy initial model may have been trained with non-open dataset

Doesn't it automatically download and use default weights if it doesn't find the D_0.pt and G_0.pt files? Is there a command line option for avoiding that or I should look into the code?

> Use a dataset that includes a variety of speakers, and is very loosely licensed, such as VCTK.

I'll make some research but that already sounds interesting

> Not sure how long it will take, but expect training to take anywhere from a few hours to a few days.

Currently training my own model for my own purposes over a 12GB dataset and it takes around 10 hrs for 1000 epochs. Expecting a week for something like VCTK, might be wrong. I'll check it out in the next days and report back.

What would be the preferred way of training? Should I use QuickVC?

Thanks in advance for your time.

**Reply by SamueleLorefice** [reply]
As this is supposed to be a Singing Voice specific model, I think I found a good dataset with CC-4-Attribution license [here](https://zenodo.org/record/1442513), It isn't really that big but it has 20 voices (9 female, 11 male) and clean samples. I gave it a listen and sounds promising. If anyone finds anything better link it to me.

**Comment by artificialguybr** [comment]
I'm pretty sure this is a dumb question, but wouldn't it be a case of using Mozilla Commom Voice? Couldn't it work with mozilla?

**Comment by BillieFrano** [comment]
What different result could we achieve after training our own initial model?
